src/main/java/de/cas/mse/exercise/diamond/Diamond.java

```diff
@@ -4,28 +4,36 @@
 
 public class Diamond {
 
-	public String print(int n) {
-		if (n <= 0 || n % 2 == 0) {
+	private static final char NEWLINE = '\n';
+	private static final char STAR = '*';
+	private static final char SPACE = ' ';
+
+	public String print(int totalLineCount) {
+		if (totalLineCount <= 0 || totalLineCount % 2 == 0) {
 			return null;
 		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
-			builder.insert(0, chars);
-			builder.append(chars);
+		StringBuilder builder = new StringBuilder(buildLine(totalLineCount, totalLineCount));
+		
+		for (int starCount = totalLineCount - 2; starCount > 0; starCount -= 2) {
+			String line = buildLine(totalLineCount, starCount);
+			builder.insert(0, line);
+			builder.append(line);
 		}
+
 		return builder.toString();
 	}
 
-	private char[] make(int i, int j) {
-		int amount = ((i - j) / 2);
-		char[] chars = new char[amount + j + 1];
-		if (amount > 0) {
-			Arrays.fill(chars, 0, amount, ' ');
+	private String buildLine(int totalLineCount, int starCount) {
+		final int blankAmount = ((totalLineCount - starCount) / 2);
+		char[] lineChars = new char[blankAmount + starCount + 1];
+		if (blankAmount > 0) {
+			Arrays.fill(lineChars, 0, blankAmount, SPACE);
 		}
-		Arrays.fill(chars, amount, amount + j, '*');
-		chars[chars.length - 1] = '\n';
-		return chars;
+		Arrays.fill(lineChars, blankAmount, blankAmount + starCount, STAR);
+		
+		final int lastIndex = lineChars.length - 1;
```

Comment:
Too much. Variable is declared in method and used right after that. Would be usefull if it was needed more then once.